### PR TITLE
[Merged by Bors] - feat: port CategoryTheory.Idempotents.KaroubiKaroubi

### PR DESCRIPTION
**Pull request by joelriou**
---
- [x] depends on: #3290
- [x] depends on: #3291

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by semorrison** [comment]
This PR/issue depends on:
* leanprover-community/mathlib4#3290
* leanprover-community/mathlib4#3291
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

**Comment by semorrison** [comment]
This PR/issue depends on:
* ~~leanprover-community/mathlib4#3290~~
* ~~leanprover-community/mathlib4#3291~~
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

File: Mathlib/CategoryTheory/Idempotents/KaroubiKaroubi.lean [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+namespace CategoryTheory
+
+namespace Idempotents
+
+namespace KaroubiKaroubi
+
+variable (C : Type _) [Category C]
+
+-- porting note: added to ease automation
+@[simp]
```
Comment (ChrisHughes24) [inline Mathlib/CategoryTheory/Idempotents/KaroubiKaroubi.lean:35]: Should this also be `reassoc`?

Comment (joelriou) [reply Mathlib/CategoryTheory/Idempotents/KaroubiKaroubi.lean]: Thanks. Yes, indeed it should. I have just added it.

**Comment by ChrisHughes24** [comment]
bors merge

**Pull request closed without merging**